packages/common/src/services/coingecko/CoingeckoClient.ts

```diff
@@ -79,6 +96,30 @@ export class CoingeckoClient {
     }
   }
 
+  async getNewCoingeckoId(
+    oldId: CoingeckoId,
+    address?: EthereumAddress,
+  ): Promise<CoingeckoId> {
+    if (address === undefined) {
+      throw new Error(
+        `Server responded with non-2XX result: Could not fetch the prices for ${oldId.toString()}`,
+      )
+    }
+    const list = await this.getCoinList({ includePlatform: true })
+    const coingeckoSupported = list.find((item) => {
+      const addr = item.platforms.ethereum
+      return (
+        addr?.toLocaleLowerCase() === address.toString().toLocaleLowerCase()
+      )
+    })
+    if (coingeckoSupported?.id === undefined) {
+      throw new Error(
+        `Server responded with non-2XX result: Could not fetch the prices for ${oldId.toString()}`,
+      )
+    }
+    return coingeckoSupported.id
```

Comment:
You are not remembering the new id, so next time you have to query it again. I'd keep an id map storing the latest ids so that you can only query once every change